Goc's goodie... airplane supply crate drop!

goranalkovic-infinum opened the pull request:
_This is gonna be a huge one, commits are your friends in this case 😄_

After many hours of work, my final ~airdrop~ goodie drop is here! Let's go through the changes.

# Component/helper changes

## New components

### `IconLabel`
A simple component that streamlines element icon labels!

![image](https://user-images.githubusercontent.com/77000136/127208286-180bc415-3491-4ab9-a10c-c6f9fcf54507.png)

Instead of this monstrosity:
```jsx
label={
    <>
        <Icon icon={icons.color} />
        {__('Your label here', 'es-fl')}
    </>
}
```
you can simply do:
```jsx
label={<IconLabel icon={icons.color} label={__('Your label here', 'es-fl')} />}
```
_Such clean, much wow!_

### `AdvancedColorPicker`
(initial version by @iruzevic and @volfkarlo, updated by me)

![image](https://user-images.githubusercontent.com/77000136/127208197-78b44f36-14a7-4c7f-be74-d71e3802e9f4.png)

Because project colors are limiting sometimes, you may need a component that would allow you to select either project colors, any solid color or gradients!

This component allows just that! It might look a bit scary because it needs quite a lot of attributes, but it's really simple to use.

Start with the manifest - you're gonna need some attributes, one for each of:
- color type
- solid color
- project color
- gradient

_Example_
```
{
	...
	"attributes": {
		"componentBgColorProject": {
			"type": "string"
		},
		"componentBgColorSolid": {
			"type": "string"
		},
		"componentBgColorGradient": {
			"type": "string"
		},
		"componentBgColorType": {
			"type": "string",
			"default": ""
		},
		...
	},
	...
	"variables": {
		"componentBgColorProject": [
			{
				"variable": {
					"component-bg-color-project": "var(--global-colors-%value%)"
				}
			}
		],
		"componentBgColorSolid": [
			{
				"variable": {
					"component-bg-color-solid": "%value%"
				}
			}
		],
		"componentBgColorGradient": [
			{
				"variable": {
					"component-bg-color-gradient": "%value%"
				}
			}
		],
	}
	...
}
```

After adding the attributes, add corresponding CSS variables, open the `component-style.scss` file and add the corresponding entries for the styles:

```css
background-color: var(--component-bg-color-solid, var(--component-bg-color-project, var(--global-colors-black));
background-image: var(--component-bg-color-gradient);
```

Almost there - now just add the component to Options. **Don't forget to pass the `globalManifest`!**

```jsx
import { checkAttr, getAttrKey, AdvancedColorPicker } from '@eightshift/frontend-libs/scripts';
import globalManifest from '../../../manifest.json';
import manifest from '../manifest.json';
...

const componentBgColorType = checkAttr('componentBgColorType', attributes, manifest);
const componentBgColorProject = checkAttr('componentBgColorProject', attributes, manifest);
const componentBgColorSolid = checkAttr('componentBgColorSolid', attributes, manifest);
const componentBgColorGradient = checkAttr('componentBgColorGradient', attributes, manifest);

...

<AdvancedColorPicker
	globalManifest={globalManifest}
	label={<IconLabel icon={icons.backgroundType} label={__('Background type', 'es-fl')} />}
	type={componentBgColorType}
	colorProject={componentBgColorProject}
	colorSolid={componentBgColorSolid}
	colorGradient={componentBgColorGradient}
	onChangeProject={
		(value) => setAttributes({
			[getAttrKey('componentBgColorProject', attributes, manifest)]: value,
			[getAttrKey('componentBgColorSolid', attributes, manifest)]: undefined,
			[getAttrKey('componentBgColorGradient', attributes, manifest)]: undefined
		})
	}
	onChangeSolid={
		(value) => setAttributes({
			[getAttrKey('componentBgColorProject', attributes, manifest)]: undefined,
			[getAttrKey('componentBgColorSolid', attributes, manifest)]: value.hex,
			[getAttrKey('componentBgColorGradient', attributes, manifest)]: undefined
		})
	}
	onChangeGradient={
		(value) => setAttributes({
			[getAttrKey('componentBgColorProject', attributes, manifest)]: undefined,
			[getAttrKey('componentBgColorSolid', attributes, manifest)]: undefined,
			[getAttrKey('componentBgColorGradient', attributes, manifest)]: value
		})
	}
	onChangeType={(value) => setAttributes({
		[getAttrKey('cardWithBackplateBgColorType', attributes, manifest)]: value,
		[getAttrKey('cardWithBackplateBgColorProject', attributes, manifest)]: undefined,
		[getAttrKey('cardWithBackplateBgColorSolid', attributes, manifest)]: undefined,
		[getAttrKey('cardWithBackplateBgColorGradient', attributes, manifest)]: undefined
	})}
/>
```

... and you're done!

Wasn't it simple? 😄  Guess it was a bit... boilerplatey, but hey, that's why the example is up there for you to copy-paste!

Closes #371.

### `AlignmentToolbar`
(initial version by @iruzevic and @volfkarlo, updated by me)

Thing aligning - streamlined!

![image](https://user-images.githubusercontent.com/77000136/127210188-9adc9832-727d-457a-9874-0d01fa70aa5e.png)

This component makes it easy to add a toolbar button for three most often used kinds of aligns:
- text align
- object horizontal align
- object vertical align

Usage is self-explainatory:

```jsx
<AlignmentToolbar
	value={componentAlign}
	options={getOption('componentAlign', attributes, manifest)}
	label={sprintf(__('%s content align', 'es-fl'), manifestTitle)}
	onChange={(value) => setAttributes({ [getAttrKey('componentAlign', attributes, manifest)]: value })}
	type={AlignmentToolbarType.HORIZONTAL}
/>
```

The types can be `AlignmentToolbarType.TEXT`, `AlignmentToolbarType.HORIZONTAL` or `AlignmentToolbarType.VERTICAL` and are defined in a object-enum, imported from the same place.

Check the doc comment for more info.

Closes #271 and #369.

### `MatrixAlignControl`
Because WordPress.

As we all know, using `__experimental` blocks is not always a good idea, but sometimes it's unavoidable. One such case is the `__experimentalBlockAlignmentMatrixToolbar`, used in many components.

Turns out that at one point between WordPress 5.7 and 5.8 the component changed... its name. Everything else is the same.

To work around that and not break everything older than 5.8, we now have a `MatrixAlignControl` which will get you the right component for the environment you have.

Just import it and replace the name of the current component - everything else* is the same, all props get passed.

(* maybe something changed in the component itself, but oh well, out of our scope 😄 )

## Component updates

### `CustomSelect`
Because I love making people's lives easier, it's now simpler than ever to convert the boring old `SelectControl` into a nice&shiny `CustomSelect` control!

We start with this:
```jsx
<SelectControl
	label={<IconLabel icon={icons.demo} label={__('Demo', 'es-fl')} />}
	value={componentDemoProp}
	options={getOption('componentDemoProp', attributes, manifest)}
	onChange={(value) => setAttributes({ [getAttrKey('componentDemoProp', attributes, manifest)]: value })}
/>
```
Now watch closely:
```jsx
<CustomSelect
	label={<IconLabel icon={icons.demo} label={__('Demo', 'es-fl')} />}
	value={componentDemoProp}
	options={getOption('componentDemoProp', attributes, manifest)}
	onChange={(value) => setAttributes({ [getAttrKey('componentDemoProp', attributes, manifest)]: value })}
	simpleValue
/>
```

💥 , done!

The `simpleValue` property (**works only with non-asnyc and single-value selects!**) will take the current value and find the correct 'full' value in the options provided so everything is fine for the select. Also, instead of returning a whole object, it just returns the value.

### `LinkToolbarButton`
Now the remove button only shows when there's a link selected.

### `ColorPickerComponent`
I put it on a diet so it has a nice new inline design, the 2 color icons close to eachother were a bit redundant.

Also now comes with dynamic fancy icons (if you select them).

This is the default new design:
![image](https://user-images.githubusercontent.com/77000136/127213069-398545ab-98fb-4dd7-9ebd-35a368a925fe.png)

And this is how it looks when you pick a color:
![image](https://user-images.githubusercontent.com/77000136/127212867-08d4d63c-463f-45b0-a76a-f00607bb7b1e.png)

And when opened:
![image](https://user-images.githubusercontent.com/77000136/127212963-80b46845-cb10-44fc-b91e-b7dadc087721.png)
(oh yeah, and the remove button is now in the popup)

Now the fun starts. What you saw there was actually a `ColorPickerType.GENERIC`.

To make it look nice in specific cases, you might opt in a for a fancy icon with a dynamic color part. The images will make it clearer:

`ColorPickerType.BACKGROUND_COLOR`
![image](https://user-images.githubusercontent.com/77000136/127213502-06e0848b-b394-449d-b22f-3f374065ddb9.png)

`ColorPickerType.TEXT_COLOR`
![image](https://user-images.githubusercontent.com/77000136/127213524-1873980e-76f8-42a3-b844-b30e716af3d1.png)

`ColorPickerType.TEXT_HIGHLIGHT_COLOR`
![image](https://user-images.githubusercontent.com/77000136/127213555-c187aa01-db78-443c-9a48-5e4348a4abbb.png)

Everything else works the same as it was until now. The `ColorPickerType.GENERIC` is the default and that will be fine for most uses.

### `HeadingLevel`
The component was rewritten, now using `OptionsPicker` in the background and some new icons to replace the SVG-path-replacing chaos before it :)

### `ServerSideRender`
For some reason the loading spinner wasn't showing (properly), that has now been fixed and we have a nice new spinner that will look nice on any background!

## Other new stuff

### Icons
There are some new UI icons (I've stopped counting at this point) and a lot of them have been refreshed (read: redrawn).

Here's the current icon sheet:
![image](https://user-images.githubusercontent.com/77000136/127213927-90e98f68-0c9b-4266-93c3-18889e8358d0.png)

There's one new block icon and some have been refreshed.

Icon sheet is as follows:
![image](https://user-images.githubusercontent.com/77000136/127213994-1e44b35b-cf49-4761-9d03-18108eeed7ba.png)

### Helpers
A `unescapeHTML` helper was added for those situations where that pesky `&#039;` makes your CustomSelect entry look ugly.

# And now... the star of the show!
Of course, the PR wouldn't be called an _airplane supply crate drop_ if it was just the changes above.

Every single* component and block was checked, fixed if needed and had its UI and UX updated and all the components and the JS logic cleaned up.
(*some things like menus will be restyled later)

Basic list of changes from the top of my head:
- new default colors, block icon colors and breakpoints in the manifest
- new default font (_Inter_ vs _Open sans_) [still not included in the project] and a new fallback - the "system font stack" (so it looks good in every platform, f**k Arial!)
- block options refreshed (see examples below)
- Lists has an option to show the color only on bullets/markers
- Accordion had a minor structure cleanup and a major style and JS logic cleanup (Closes #413)
- Carousel now has options to add prev/next buttons and pagination
- Featured categories and Featured posts are now way less confusing to use :smile: They also use CSS vars now (closes #355 and #356)
- Columns were rewritten to use CSS Grid and variables (closes #353 and #354)
- Wrapper was cleaned up and its control made cleaner
- Basic components like Heading, Paragraph, Button, Lists now have a bit more color and font size options by default (and some color combos are pretty fun I might say)
- Replaced alignment toolbars on text controls (closes #271)
- (this was already done, but just so it links the issue) - Swiper removal (closes #325)
...

For more check the PRs and also it would be cool if you made a new project (with this feature/ branch for now) and tried stuff out a bit so I can fix issues if they pop up.

This is already a very long changelog, so to end it, below are some freshly-redesigned panels.

🥂  to the fresh new-old Frontend Libs! 😄 

![image](https://user-images.githubusercontent.com/77000136/127216306-ef68952c-0cf1-49a7-95fd-fd874786420f.png)

iruzevic commented:
@infinum/wordpress-team If you want to test everything use this commands:

```shell
wp boilerplate setup_theme
wp boilerplate use_component --name='all'
wp boilerplate use_block --name='all'
```

This will copy everything to your project and you can use this PR #409 for importing all in DOM for testing

iruzevic commented:
Issues:
1. If you select button width block and then click Display as link it will remove width option but attribute will remain set to block

goranalkovic-infinum commented:
@iruzevic 

All the editor issues have been fixed.

From front issues:
1. fixed
2. fixed by https://github.com/infinum/eightshift-boilerplate/pull/248
3. looks fine to me now, maybe something in the meantime fixed it :D
4. yup, the images are behaving kinda weird, I'll reasearch it a bit

goranalkovic-infinum commented:
@infinum/wordpress-team 

For now, for testing you can use
```
npx create-wp-project --eightshiftLibsBranch="release/3.1.0" --eightshiftFrontendLibsBranch="feature/boilerplate-blocks-ux-updates" --eightshiftBoilerplateBranch="feature/fixes-for-fl-v5"
```
while creating your theme to get all the boilerplate fixes as well.

IvanGrginovInf reviewed (changes requested):
Awesome work, don't know what else to say...nice! Just please fix the failing build checks.

iruzevic reviewed (changes requested):
A few small suggestions

iruzevic commented:
@goranalkovic-infinum 
# Additional improvements Storybook:
1. Button component - Options - missing labels in the Size and Width
2. Drawer - can we make something so it is visible?

dingo-d commented on scripts/components/advanced-color-picker/advanced-color-picker.js:
```diff
@@ -0,0 +1,128 @@
+
+const CustomPickerOption = props => {
+	return (
+		<components.Option {...props}>
```
Can these be extracted somehow? 

All those parts of React Select components are using `export default`. We could then try to import them `as ReactSelectOption` etc. just not to confuse them with our own when reading the code.

dingo-d commented on scripts/components/advanced-color-picker/advanced-color-picker.js:
```diff
@@ -0,0 +1,128 @@
+
+		types = [
+			{
+				"label": "No background",
```
These labels cannot be translated?

goranalkovic-infinum replied:
They can be, added, thanks! 😄

dingo-d commented on scripts/components/advanced-color-picker/advanced-color-picker.js:
```diff
@@ -0,0 +1,128 @@
+const CustomPickerOption = props => {
+	return (
+		<components.Option {...props}>
+			<span className='es-custom-select-flex'>{getIcon(props.value)} {props.label}</span>
```
I'd also extract the props out

```js
const {...} = props;
```

dingo-d commented on scripts/components/advanced-color-picker/advanced-color-picker.js:
```diff
@@ -0,0 +1,128 @@
+	}
+}
+
+const CustomPickerOption = props => {
```
```suggestion
const CustomPickerOption = (props) => {
```

dingo-d commented on scripts/components/color-picker-component/color-picker-component.js:
```diff
@@ -1,22 +1,36 @@
+ * @param {React.Component?} props.label                - Label to represent the control
+ * @param {boolean} [props.canReset=true]               - If `true`, a clear/reset button is shown.
+ * @param {string} [props.pickerPopupTitle]             - Color picker popup title.
+ * @param {string} [props.resetLabel]                 - 'Reset' button tooltip.
```
Spacing is off here.

dingo-d commented on scripts/components/color-picker-component/color-picker-component.js:
```diff
@@ -25,9 +39,8 @@ export const ColorPickerComponent = ({
-	editColorLabel = __('Change color'),
-	addColorLabel = __('Pick a color'),
-	resetTootlip = __('Reset'),
+	resetLabel = __('Reset'),
```
Missing text domain here

dingo-d commented on scripts/components/alignment-toolbar/alignment-toolbar.js:
```diff
@@ -0,0 +1,140 @@
+	const textAligns = {
+		left: showAlignStart ? {
+			icon: icons.textAlignLeft,
+			title: __('Align text left'),
```
Missing text domains in the file

dingo-d commented on scripts/components/alignment-toolbar/alignment-toolbar.js:
```diff
@@ -0,0 +1,140 @@
+	const showAlignStart = options.includes('left');
+	const showAlignCenter = options.includes('center');
+	const showAlignEnd = options.includes('right');
+	const showAlignJustify =  options.includes('justify');
+	const showAlignStretch =  options.includes('stretch');
```
```suggestion
	const showAlignJustify = options.includes('justify');
	const showAlignStretch = options.includes('stretch');
```

dingo-d commented on scripts/components/alignment-toolbar/docs/readme.mdx:
```diff
@@ -0,0 +1,17 @@
+<AlignmentToolbar
+	value={blockquoteContentAlign}
+	options={getOptions(manifest, componentName, 'contentAlign', options)}
+	label={__('Content align', 'productive')}
```
productive text domain 😄

dingo-d commented on blocks/init/src/Blocks/custom/button/components/button-toolbar.js:
```diff
@@ -23,10 +22,10 @@ export const ButtonToolbar = ({ attributes, setAttributes }) => {
 				value={buttonAlign}
 				options={getOption('buttonAlign', attributes, manifest)}
-				label={sprintf(__('%s button align', 'eightshift-frontend-libs'), manifestTitle)}
+				label={sprintf(__('%s align', 'newboilerplate'), manifestTitle)}
```
newboilerplate?

goranalkovic-infinum replied:
dev-theme name, whoops 😄

dingo-d commented on blocks/init/src/Blocks/components/copyright/manifest.json:
```diff
@@ -5,7 +5,7 @@
 		"attributes": {
 			"copyrightBy": "Eightshift",
-			"copyrightYear": "2020",
+			"copyrightYear": "2021",
```
When in Rome... Can this be removed from the manifest? And just rendered out in PHP/JS so that we don't need to change the year manually?

iruzevic replied:
this is just an example for storybook. No need to change this we can put 1970 just for fun :D

goranalkovic-infinum replied:
The default footer renders the current year, I'll remove the example attr, but leave the attribute though, some websites like to have a static (C) year.

dingo-d commented on blocks/init/src/Blocks/components/heading/components/heading-options.js:
```diff
@@ -29,48 +29,34 @@ export const HeadingOptions = (attributes) => {
-									{__('Color', 'eightshift-frontend-libs')}
-								</>
-							}
+							label={<IconLabel icon={icons.color} label={__('Color', 'newboilerplate')} />}
```
newboilerplate

dingo-d commented on blocks/init/src/Blocks/components/heading/components/heading-options.js:
```diff
@@ -29,48 +29,34 @@ export const HeadingOptions = (attributes) => {
-								</>
-							}
+						<CustomSelect
+							label={<IconLabel icon={icons.textSize} label={__('Font size', 'newboilerplate')} />}
```
⬆️

goranalkovic-infinum commented on blocks/init/src/Blocks/components/loader/components/loader-editor.js:
```diff
@@ -24,17 +24,24 @@ export const LoaderEditor = (attributes) => {
-			}
-		</>
+		<div className={loaderClass}>
+			<svg xmlns="http://www.w3.org/2000/svg" width="44" height="44" viewBox="0 0 44 44" >
```
Sure!

dingo-d commented on blocks/init/src/Blocks/custom/heading/components/heading-toolbar.js:
```diff
@@ -23,10 +22,9 @@ export const HeadingToolbar = ({ attributes, setAttributes }) => {
 				value={headingAlign}
 				options={getOption('headingAlign', attributes, manifest)}
-				label={sprintf(__('%s text align', 'eightshift-frontend-libs'), manifestTitle)}
+				label={sprintf(__('%s text align', 'newboilerplate'), manifestTitle)}
```
⬆️

dingo-d commented on scripts/components/matrix-align-control/matrix-align-control.js:
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { __experimentalBlockAlignmentMatrixControl as BlockAlignmentMatrixControl } from '@wordpress/block-editor';
+import { __experimentalBlockAlignmentMatrixToolbar as BlockAlignmentMatrixToolbar } from '@wordpress/block-editor';
```
This will work in all cases? Like in 5.8 they use one in 5.7 they use another, but this won't throw some undefined import error if this is missing.

I think in the docs we should just note that we need to look when they stop making this as experimental 😅

goranalkovic-infinum replied:
There's only a warning in Storybook, no errors in the browser as you never try to use the non-existant component.

dingo-d commented on blocks/init/src/Blocks/components/image/components/image-options.js:
```diff
@@ -34,91 +35,90 @@ export const ImageOptions = (attributes) => {
 					{showImageAlt &&
 						<TextareaControl
-							label={__('Alt text', 'eightshift-frontend-libs')}
+							label={<IconLabel icon={icons.altText} label={__('Alt text', 'newboilerplate')} />}
```
newboilerplate

dingo-d commented on blocks/init/src/Blocks/components/image/components/image-options.js:
```diff
@@ -34,91 +35,90 @@ export const ImageOptions = (attributes) => {
-							help={__('If checked, the image will always stretch the full width of the container and ignore its maximum width.', 'eightshift-frontend-libs')}
+						<IconToggle
+							icon={icons.size}
+							label={__('Fill container', 'newboilerplate')}
```
⬆️

dingo-d commented on blocks/init/src/Blocks/custom/image/components/image-toolbar.js:
```diff
@@ -1,15 +1,14 @@
-		<BlockAlignmentMatrixToolbar
-			label={__('Image Position', 'eightshift-frontend-libs')}
+		<MatrixAlignControl
+			label={__('Image position', 'newboilerplate')}
```
text domain

dingo-d commented on blocks/init/src/Blocks/components/lists/components/lists-editor.js:
```diff
@@ -18,7 +18,7 @@ export const ListsEditor = (attributes) => {
 		blockClass,
 		additionalClass,
-		placeholder = __('Add Content', 'eightshift-frontend-libs'),
+		placeholder = __('Add content', 'newboilerplate'),
```
text-domain

dingo-d commented on blocks/init/src/Blocks/components/lists/components/lists-options.js:
```diff
@@ -26,51 +27,47 @@ export const ListsOptions = (attributes) => {
-									{__('Color', 'eightshift-frontend-libs')}
-								</>
-							}
+							label={<IconLabel icon={icons.color} label={__('Color', 'newboilerplate')} />}
```
text domain

dingo-d commented on blocks/init/src/Blocks/custom/lists/components/lists-toolbar.js:
```diff
@@ -15,7 +14,8 @@ export const ListsToolbar = ({ attributes, setAttributes }) => {
 			value={listsAlign}
 			options={getOption('listsAlign', attributes, manifest)}
-			label={sprintf(__('%s text align', 'eightshift-frontend-libs'), manifestTitle)}
+			label={sprintf(__('%s text align', 'newboilerplate'), manifestTitle)}
```
text domain

dingo-d commented on blocks/init/src/Blocks/components/page-overlay/page-overlay-style.scss:
```diff
@@ -13,7 +13,11 @@
+		opacity: 0.6;
 		pointer-events: auto;
+
+		@supports (mix-blend-mode: multiply) {
```
Cool `@supports` 👍🏼

dingo-d commented on blocks/init/src/Blocks/components/paragraph/components/paragraph-editor.js:
```diff
@@ -11,15 +11,14 @@ export const ParagraphEditor = (attributes) => {
 		blockClass,
 		additionalClass,
-		placeholder = __('Add Content', 'eightshift-frontend-libs'),
+		placeholder = __('Add content', 'newboilerplate'),
```
text domain

dingo-d commented on blocks/init/src/Blocks/components/paragraph/components/paragraph-options.js:
```diff
@@ -14,63 +13,50 @@ export const ParagraphOptions = (attributes) => {
-									{__('Color', 'eightshift-frontend-libs')}
-								</>
-							}
+							label={<IconLabel icon={icons.color} label={__('Color', 'newboilerplate')} />}
```
text-domain

dingo-d commented on blocks/init/src/Blocks/custom/paragraph/components/paragraph-toolbar.js:
```diff
@@ -15,7 +14,8 @@ export const ParagraphToolbar = ({ attributes, setAttributes }) => {
 			value={paragraphAlign}
 			options={getOption('paragraphAlign', attributes, manifest)}
-			label={sprintf(__('%s text align', 'eightshift-frontend-libs'), manifestTitle)}
+			label={sprintf(__('%s text align', 'newboilerplate'), manifestTitle)}
```
text domain

dingo-d commented on blocks/init/src/Blocks/components/card/components/card-toolbar.js:
```diff
@@ -41,8 +40,9 @@ export const CardToolbar = (attributes) => {
 				value={cardAlign}
 				options={getOption('cardAlign', attributes, manifest)}
-				label={sprintf(__('%s text align', 'eightshift-frontend-libs'), manifestTitle)}
+				label={sprintf(__('%s text align', 'newboilerplate'), manifestTitle)}
```
text-domain

dingo-d commented on blocks/init/src/Blocks/components/accordion/components/accordion-editor.js:
```diff
@@ -10,48 +10,57 @@ export const AccordionEditor = (attributes) => {
 		additionalClass,
 		blockClass,
-		placeholder = __('Add Content', 'eightshift-frontend-libs'),
+		placeholder = __('Add content', 'newboilerplate'),
```
text domain

dingo-d commented on blocks/init/src/Blocks/components/accordion/components/accordion-options.js:
```diff
@@ -1,16 +1,16 @@
 	const {
 		setAttributes,
-		label = __('Accordion', 'eightshift-frontend-libs'),
+		label = __('Accordion', 'newboilerplate'),
```
Text domain

dingo-d commented on blocks/init/src/Blocks/components/share/components/share-editor.js:
```diff
@@ -27,17 +29,16 @@ export const ShareEditor = (attributes) => {
-			}
-		</>
+		<div className={shareClass}>
+			{__('Share on', 'newboilerplate')}
```
I think I'll stop commenting these, just do a search-replace 😂

dingo-d commented on blocks/init/src/Blocks/components/share/share.php:
```diff
@@ -12,7 +12,7 @@
 $postUrl = $attributes['postUrl'] ?? \get_the_permalink();
 $postTitle = $attributes['postTitle'] ?? \get_the_title();
-$postImageUrl = $attributes['postImageUrl'] ?? \get_the_post_thumbnail_url(\get_the_ID(), 'large');
+$postImageUrl = $attributes['postImageUrl'] ?? \get_the_post_thumbnail_url(\get_the_ID(), 'large') ?? '';
```
Will this work as intended?

In the case the `$attributes['postImageUrl']` exists, it will show it. Then if previous is false, and  `get_the_post_thumbnail_url` works it shows that url, and fallback to '' in the end?

goranalkovic-infinum replied:
Yup, see: http://sandbox.onlinephpfunctions.com/code/eaf827f122c0662fa148861c5d221e55c28f4d11

dingo-d replied:
But we're ok with that? 😄

goranalkovic-infinum replied:
Yup

dingo-d commented on blocks/init/src/Blocks/components/social-links/components/social-links-editor.js:
```diff
@@ -24,31 +23,26 @@ export const SocialLinksEditor = (attributes) => {
+						href={href}
+						title={title}
+						dangerouslySetInnerHTML={{ __html: manifest.icons[icon] }}
+						target="_blank" rel="nofollow noreferrer noopener"
```
Do we want this by default? I think there is a legit use case for these attributes, but sometimes they could hurt the SEO iirc

goranalkovic-infinum replied:
@infinum/wordpress-team chime in, please

IvanGrginovInf replied:
`nofollow` should generally only be used for:
- Ads
- Paid links
- If you're linking to something shady xD

So here I don't think we should have a `nofollow` a branded social link is a perfectly valid link IMO.

IvanGrginovInf replied:
`noreferrer noopener` is a security-issue fix for links opening in new tab so this should naturally stay.

goranalkovic-infinum replied:
Fixed!

dingo-d reviewed (changes requested):
Left some comments. This was a massive overhaul! Hats off 👍🏼

mbmjertan commented on blocks/init/src/Blocks/components/accordion/accordion.php:
```diff
@@ -24,6 +24,7 @@
 $accordionTitle = Components::checkAttr('accordionTitle', $attributes, $manifest);
 $accordionContent = Components::checkAttr('accordionContent', $attributes, $manifest);
```
You have a variable name mismatch with the accordion block here, as it gets an `accordionAccordionContent` attribute. This results in accordion having no content output on frontend.

```php
echo Components::render( // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	'accordion',
	Components::props('accordion', $attributes, [
		'accordionAccordionContent' => $innerBlockContent
	])
);
```

mbmjertan reviewed (changes requested):
Apart from what I commented for the accordion block, I noticed some issues with the carousel block:
* When enabled, navigation buttons don't work.  
* Image blocks that are children of a carousel block are scaled ignoring their aspect ratio (they have width and height set to 100%)
![Screenshot 2021-07-28 at 13 15 40](https://user-images.githubusercontent.com/1742806/127314906-8aa0c926-d737-4d9b-9c2b-cb7a168075a4.png)

Amazing work! 🎉

goranalkovic-infinum commented:
@iruzevic the images should behave way better now, especially if 'Stretch to fill' is not enabled. Please let me know if they're good enough now.

mbmjertan commented:
Seems like the issues with carousel and accordion have been fixed :)

Although, this doesn't seem to fix #413 like you said: 
https://github.com/infinum/eightshift-frontend-libs/blob/8d821a2e245102682fc9ea397dfe9603375d93ce/blocks/init/src/Blocks/components/accordion/assets/index.js#L12-L16

goranalkovic-infinum commented:
> Seems like the issues with carousel and accordion have been fixed :)
> 
> Although, this doesn't seem to fix #413 like you said:
> https://github.com/infinum/eightshift-frontend-libs/blob/8d821a2e245102682fc9ea397dfe9603375d93ce/blocks/init/src/Blocks/components/accordion/assets/index.js#L12-L16

Good point (I at the first read thought there were imports in a loop 😂 ), but I don't see multiple elements sharing the same one class logic being any good.

These _are_ class instances, but classes in JS are basically functions with a lot of syntactic sugar on top.

You would end up using a loop somewhere anyways, and if you take Carousel for example, that would get messy pretty quickly with all the Swiper logic and the "state" of each element.

Just my $0.02

iruzevic commented:
@goranalkovic-infinum I have left additional comments to fix here:

https://github.com/infinum/eightshift-frontend-libs/pull/414#issuecomment-888067208

> Card - There is difference in spacing between heading, paragraph, image and button in editor and on the frontend.
and also an issue on the card inside the featured-posts/categories block. Because the content is rendered from the frontend once rendered using ssr compnente in the admin this card heading and paragraph will get h4 and p tags and that causes the problem. You can fix this by overriding the tags once component is server side rendered. Not sure why this isn't working but this part will fix the issue https://github.com/infinum/eightshift-frontend-libs/blob/d58ac9f69030af08e9bb43ddc8fe6cb4deabb6cf/blocks/init/src/Blocks/custom/featured-categories/featured-categories.php#L58 here is an fix from the current main branch

volfkarlo commented on blocks/init/src/Blocks/components/lists/manifest.json:
```diff
@@ -95,7 +127,116 @@
+					"variable": {
+						"lists-color": "var(--global-colors-cream)",
+						"lists-link-color": "var(--global-colors-yellow)",
+						"lists-link-color-hover": "var(--global-red)"
```
```suggestion
						"lists-link-color-hover": "var(--global-colors-red)"
```

volfkarlo commented on blocks/init/src/Blocks/components/lists/lists-style.scss:
```diff
@@ -1,26 +1,32 @@
 		text-decoration: underline;
-		color: var(--lists-link-color);
+		color: var(--lists-link-color, var(--global-colors-primary, #5555CE));
+		text-decoration-color: var(--lists-link-color, var(--global-colors-primary, #5555CE));
```
Why not something like this instead of this literals?

```suggestion
		text-decoration-color: var(--lists-link-color, var(--global-colors-primary, var(--global-colors-yellow));
```

goranalkovic-infinum replied:
I wanted the default to be the "primary" color, if you don't have that one then you probably don't have the "yellow" either 😄  Just a hard fallback and can be removed if you think it's extra

volfkarlo commented on blocks/init/src/Blocks/components/loader/loader.php:
```diff
@@ -26,14 +26,11 @@
 	Components::selector($additionalClass, $additionalClass),
-	Components::selector($loaderUseOverlay, $componentClass, '', 'use-overlay'),
+	// @phpstan-ignore-next-line
+	Components::selector($loaderUseOverlay, $componentClass, '', 'use-overlay'), Components::selector($loaderUseOverlay, $componentClass, '', 'use-overlay'),
```
why double?

```suggestion
	Components::selector($loaderUseOverlay, $componentClass, '', 'use-overlay'), 
```

goranalkovic-infinum replied:
Slow mac + lunch arriving 😄 
Fixed, thanks!

volfkarlo commented on blocks/init/src/Blocks/components/page-overlay/components/page-overlay-editor.js:
```diff
@@ -22,11 +22,9 @@ export const PageOverlayEditor = (attributes) => {
+		return null;
+	}
+
+	return (<div className={overlayClass}></div>);
```
```suggestion
	return (
		<div className={overlayClass}></div>
	);
```

```suggestion
	return <div className={overlayClass}></div>;
```

It so doesn't matter but just to have it

volfkarlo commented on blocks/init/src/Blocks/components/paragraph/components/paragraph-options.js:
```diff
@@ -1,7 +1,6 @@
-import { ColorPaletteCustom, checkAttr, getAttrKey, icons, getOption } from '@eightshift/frontend-libs/scripts';
-import { SelectControl, Icon, ToggleControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { ColorPaletteCustom, checkAttr, getAttrKey, icons, getOption, ComponentUseToggle, IconLabel, CustomSelect } from '@eightshift/frontend-libs/scripts';
```
probably not needed comment but it would be cleaner if we would do it something like this

```suggestion
import { ColorPaletteCustom, ComponentUseToggle, IconLabel, CustomSelect, checkAttr, getAttrKey, icons, getOption } from '@eightshift/frontend-libs/scripts';
```

volfkarlo commented on blocks/init/src/Blocks/custom/accordion/components/accordion-editor.js:
```diff
@@ -8,7 +8,7 @@ export const AccordionEditor = ({ attributes, setAttributes }) => {
 			{...props('accordion', attributes, {
 				setAttributes: setAttributes,
-				accordionAccordionContent: <InnerBlocks />
+				accordionContent: <InnerBlocks />
```
This should be ok if you are receiving 
`content` in `component/accordion` but since you are suppose to get `accordionContent` I believe this is not working as it should.

goranalkovic-infinum replied:
As I explained in a comment above, becuase this is a (weird edge case and a) component, and components usually don't have `InnerBlocks`, we have to pass this directly to the component's editor, that's why it's `accordionContent` and not `accordionAccordionContent`.

volfkarlo commented on blocks/init/src/Blocks/custom/column/column-editor.scss:
```diff
@@ -1,10 +1,15 @@
-	@include column;
+	transition: {
+		// stylelint-disable-next-line declaration-no-important
+		property: opacity, transform !important;
```
how come you use important?

goranalkovic-infinum replied:
Have to override `opacity` from some default Gutenberg style.

volfkarlo commented on blocks/init/src/Blocks/custom/column/manifest.json:
```diff
@@ -167,5 +180,106 @@
+				}
+			}
+		],
+		"columnVerticalAlign": {
```
I've added horizontal align in Infinum web project. maybe it would be good to do the same here.

volfkarlo commented on blocks/init/src/Blocks/custom/columns/manifest.json:
```diff
@@ -142,5 +142,23 @@
+				}
+			}
+		],
+		"columnsGutter": [
```
just a suggestion but potentially to use name such as vertical and horizontal spacing since they both represent gap(column and row)😄

volfkarlo commented on scripts/components/custom-select/custom-select.js:
```diff
@@ -25,6 +25,8 @@ import AsyncSelect from "react-select/async";
  * @param {string} [props.sortAxis=y]                                  - If multiple-select mode is active, determines the axis the items can be sorted on. Can be `x`, `y` or `xy`.
  * @param {React.Component?} [props.customOptionComponent]             - If provided, this control replaces the default option displayed in the dropdown.
  * @param {React.Component?} [props.customSingleValueDisplayComponent] - If provided and in single-select mode, this control replaces the default current value display when the dropdown is closed.
+ * @param {React.Component?} [props.customIndicatorSeparator] 		   - If provided, adds a separator between the select content and the dropdown arrow on the right.
```
Indent of the comment.

volfkarlo commented on scripts/components/icon-label/icon-label.js:
```diff
@@ -0,0 +1,27 @@
+/**
+ * A simple icon-label combo for streamlined components.
+ * 
+ * @param {React.Component} props.icon       - Icon to display.
+ * @param {React.Component} props.icon       - Icon to display.
```
double doc, I assume it should be
```suggestion
 * @param {React.Component} props.label      - Label to display.
 * @param {React.Component} props.icon       - Icon to display.
```

volfkarlo commented on scripts/components/matrix-align-control/docs/story.js:
```diff
@@ -0,0 +1,24 @@
+	const [align, setAlign] = useState('center center');
+
+	return (
+		<MatrixAlignControl
+		label='Alignment position'
+		value={align}
+		onChange={(value) => setAlign(value)}
+	/>
```
indent

volfkarlo reviewed (changes requested):
Just a few minor changes.

Awesome work @goranalkovic-infinum 👏 🎉 🥇

goranalkovic-infinum commented:
> @goranalkovic-infinum
> 
> # Additional improvements Storybook project:
> 1. Button component - Options - missing labels in the Size and Width
> 2. Drawer - can we make something so it is visible?
> 3. Hamburger - can we make so it is visible in storybook always?
> 4. Video component - Options - Advanced the same as on Button
> 
> # Improvements storybook components
> 1. AdvancedColorPicker - icons are cut on top and bottom
> 2. ColorPickerComponent - doesn't show any thing
> 3. CustomSelect - when using Async, once the data is loaded the dropdown closes for some reason
> 4. SimpleVerticalSingleSelect - Missing labels

@iruzevic 
Additional improvements
1.
![image](https://user-images.githubusercontent.com/77000136/127500000-6c2c7cbc-dba0-4799-86c7-dec044ae3d82.png)
Buttons show the values inside the `content` attribute, do we need extra ones?

2. Done!
3. Done!
4. The advanced \/ arrow button is not supposed to have a label


Improvements storybook components
1. Fixed
2. It was filling the whole width and looked weird, reduced now
3. Not sure how to fix that, probably some state being set somewhere that causes everything to reload.
4. Fixed

goranalkovic-infinum commented on blocks/init/src/Blocks/components/accordion/accordion.php:
```diff
@@ -23,8 +23,9 @@
 
 $accordionTitle = Components::checkAttr('accordionTitle', $attributes, $manifest);
-$accordionContent = Components::checkAttr('accordionContent', $attributes, $manifest);
+$accordionContent = Components::checkAttr('accordionAccordionContent', $attributes, $manifest);
```
It's not working with `accordionContent` 🤷 
Could it be a weird interaction between <InnerBlocks /> being passed from the block to the component?
